src/Mvc/Mvc.RazorPages/src/DependencyInjection/PageConventionCollectionExtensions.cs

```diff
@@ -289,13 +337,23 @@ public static PageConventionCollection AuthorizeAreaPage(this PageConventionColl
                 throw new ArgumentException(Resources.ArgumentCannotBeNullOrEmpty, nameof(pageName));
             }
 
-            var authorizeFilter = new AuthorizeFilter(policy);
-            conventions.AddAreaPageApplicationModelConvention(areaName, pageName, model => model.Filters.Add(authorizeFilter));
+            var authorizeAttribute = new AuthorizeAttribute(policy);
```

Comment:
is this attribute instance used anywhere? the endpoint metadata is newing another one up below